public-api/src/main/java/org/pmiops/workbench/publicapi/CdrVersionsController.java

```diff
@@ -0,0 +1,72 @@
+package org.pmiops.workbench.publicapi;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableSet;
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import javax.inject.Provider;
+import org.pmiops.workbench.config.WorkbenchConfig;
+import org.pmiops.workbench.db.dao.CdrVersionDao;
+import org.pmiops.workbench.db.model.CdrVersion;
+import org.pmiops.workbench.db.model.CommonStorageEnums;
+import org.pmiops.workbench.exceptions.ServerErrorException;
+import org.pmiops.workbench.model.CdrVersionListResponse;
+import org.pmiops.workbench.model.DataAccessLevel;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class CdrVersionsController implements CdrVersionsApiDelegate {
+
+  @VisibleForTesting
+  static final Function<CdrVersion, org.pmiops.workbench.model.CdrVersion> TO_CLIENT_CDR_VERSION =
+      new Function<CdrVersion, org.pmiops.workbench.model.CdrVersion>() {
+        @Override
+        public org.pmiops.workbench.model.CdrVersion apply(CdrVersion cdrVersion) {
+          return new org.pmiops.workbench.model.CdrVersion()
+              .cdrVersionId(String.valueOf(cdrVersion.getCdrVersionId()))
+              .creationTime(cdrVersion.getCreationTime().getTime())
+              .dataAccessLevel(cdrVersion.getDataAccessLevelEnum())
+              .name(cdrVersion.getName());
+        }
+      };
+
+  private final CdrVersionDao cdrVersionDao;
+  private final Provider<WorkbenchConfig> workbenchConfigProvider;
+
+  @Autowired
+  CdrVersionsController(CdrVersionDao cdrVersionDao, Provider<WorkbenchConfig> workbenchConfigProvider) {
+    this.cdrVersionDao = cdrVersionDao;
+    this.workbenchConfigProvider = workbenchConfigProvider;
+  }
+
+
+  @Override
+  public ResponseEntity<CdrVersionListResponse> getCdrVersions() {
+    // We return CDR versions for just registered CDR versions; controlled CDR data is currently
+    // out of scope for the data browser.
+    List<CdrVersion> cdrVersions = cdrVersionDao
+        .findByDataAccessLevelInOrderByCreationTimeDescDataAccessLevelDesc(
+            ImmutableSet.of(CommonStorageEnums.dataAccessLevelToStorage(DataAccessLevel.REGISTERED)));
```

Comment:
Wouldn't we want UNREGISTERED as well?

Reply:
So we're using DataAccessLevel for CDR versions and *also* for users. For CDR versions, my assumption was the only valid values are REGISTERED and CONTROLLED -- no CDR versions are accessible to people who haven't gotten REGISTERED data access. For users, we use UNREGISTERED to represent users who haven't gotten data access yet.

(We could consider having two separate enums here to make this clearer.)

But anyway, I think REGISTERED is all we want here.